lock_api/src/remutex.rs

```diff
@@ -50,6 +51,7 @@ impl<R: RawMutex, G: GetThreadId> RawReentrantMutex<R, G> {
     #[inline]
     fn lock_internal<F: FnOnce() -> bool>(&self, try_lock: F) -> bool {
         let id = self.get_thread_id.nonzero_thread_id();
+        // REVIEW: debug assert that `id` isn't zero?
```

Comment:
Should not be needed if we go through with using `NonZeroUsize`, right? That is being proposed over in https://github.com/Amanieu/parking_lot/pull/148

Reply:
Seems reasonable!